Description: Fix compilation with wxWidgets versions <3.1
Author: David Hart <david@codelite.org>
Forwarded: no
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/codelite_terminal/TextView.cpp
+++ b/codelite_terminal/TextView.cpp
@@ -12,7 +12,11 @@ TextView::TextView(wxWindow* parent, wxWindowID winid)
     m_ctrl->SetCaretStyle(wxSTC_CARETSTYLE_BLOCK);
     m_ctrl->SetYCaretPolicy(wxSTC_CARET_STRICT | wxSTC_CARET_SLOP, 4);
     m_ctrl->SetLexer(wxSTC_LEX_CONTAINER);
+#if wxCHECK_VERSION(3, 1, 1)
     m_ctrl->StartStyling(0);
+#else
+    m_ctrl->StartStyling(0, 0x1f);
+#endif
     m_ctrl->SetWrapMode(wxSTC_WRAP_CHAR);
 #else
     m_ctrl = new TextCtrl_t(this, wxID_ANY, "", wxDefaultPosition, wxDefaultSize,
@@ -61,7 +65,11 @@ void TextView::SetDefaultStyle(const wxTextAttr& attr)
 {
 #if USE_STC
     m_defaultAttr = attr;
+#if wxCHECK_VERSION(3, 1, 1)
     m_ctrl->StartStyling(m_ctrl->GetLastPosition());
+#else
+    m_ctrl->StartStyling(m_ctrl->GetLastPosition(), 0x1f);
+#endif
 #else
     m_ctrl->SetDefaultStyle(attr);
 #endif
@@ -210,6 +218,10 @@ void TextView::Clear()
 #if USE_STC
     m_ctrl->ClearAll();
     m_ctrl->ClearDocumentStyle();
+#if wxCHECK_VERSION(3, 1, 1)
     m_ctrl->StartStyling(0);
+#else
+    m_ctrl->StartStyling(0, 0x1f);
+#endif
 #endif
 }
